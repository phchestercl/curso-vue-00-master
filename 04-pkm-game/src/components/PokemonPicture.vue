<template>
    <div class="pokemon-container">
        <img :src="imgSrc"
              class="hidden-pokemon"
              alt="pokemon">
        <img v-if="showPokemon"
                :src="imgSrc" 
                class="fade-in"
                alt="pokemon">
        
    </div>

</template>

<script>
export default {
    name:'PokemonPicture',
    props:{
        pokemonID:{ type:Number, required:true, default:1},
        showPokemon:{ type:Boolean, required:true,default:false}
    },
   /*  data(){
        return {
            pokemonID:this.pokemonID,
            }
    }, */
    watch:{

    },

    computed:{
        imgSrc(){
            return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonID}.svg`
        }
    }

}
</script>

<style scope>
/* Pokemon Picture */
.pokemon-container {
    height: 200px;
}
img {
    height: 200px;
    position: absolute;
    right: 32%;
    user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    -webkit-user-drag: none;
    -webkit-user-select: none;
}
.hidden-pokemon {
    filter: brightness(0);
}
</style>