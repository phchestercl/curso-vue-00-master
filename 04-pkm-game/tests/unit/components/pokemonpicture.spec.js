import {shallowMount} from '@vue/test-utils'
import PokemonPicture from '@/components/PokemonPicture'

describe('Pokemon Picture',()=>{
    /* let wrapper;
    //El beforeEach nos permiete inicializar el entorno antes de ejecutar los test
    beforeEach(() => {
        wrapper = shallowMount(PokemonPicture)
    }) */
    test('Match con snapshot',()=>{
        const wrapper=shallowMount(PokemonPicture,{
            props:{pokemonID:100,showPokemon:false}
        })
        expect( wrapper.html() ).toMatchSnapshot()
    })
    test ('debe mostrarla imagen oculta y el pokemon 100',()=>{
        const wrapper=shallowMount(PokemonPicture,{
            props:{ pokemonID:100, showPokemon:false}
        })
        //console.log(wrapper.html())
        const [img1, img2] = wrapper.findAll('img')
        expect(img1.exists()).toBeTruthy()
        expect(img2).toBeUndefined()
        expect(img1.classes('hidden-pokemon')).toBeTruthy()
        //console.log(wrapper.vm.imgSrc)
        const src = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg`
        expect(wrapper.vm.imgSrc).toBe(src)
        //console.log(img1.attributes('src'))
        expect(img1.attributes('src')).toBe(src)


    })
    test('Debe mostrar el pokemon si showPokemon:true',()=>{
        const wrapper=shallowMount(PokemonPicture,{
            props:{ pokemonID:100, showPokemon:true}
        })
        //console.log(wrapper.html())
        
        const img = wrapper.find('img')
    
        expect(img.exists()).toBeTruthy()
        expect(img.classes("hidden-pokemon")).toBeFalsy()
        expect(img.classes('fade-in')).toBeTruthy()
    })
})