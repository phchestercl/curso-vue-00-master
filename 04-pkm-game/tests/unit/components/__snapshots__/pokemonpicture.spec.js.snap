// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Picture Match con snapshot 1`] = `<div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg" class="hidden-pokemon" alt="pokemon"></div>`;
